cmd/exporter.go

```diff
@@ -150,54 +150,27 @@ func NewExporter(config ExporterConfig) (*Exporter, error) {
 func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	ctx := context.Background()
 
-	if e.provider == provider.WattTime {
-		result, err := e.client.GetCarbonIntensity(ctx, e.location)
-		if err != nil {
-			log.Printf("failed to get carbon intensity %#v", err)
-		}
+	result, err := e.client.GetCarbonIntensity(ctx, e.location)
+	if err != nil {
+		log.Printf("failed to get carbon intensity %#v", err)
+	}
 
-		for _, data := range result {
-			if data.MetricType == provider.AbsoluteMetricType {
-				ch <- prometheus.MustNewConstMetric(
-					marginalDesc,
-					prometheus.GaugeValue,
-					data.Value,
-					data.Location,
-					e.node,
-					data.Provider,
-					e.region,
-					data.Units,
-				)
-			}
-			if data.MetricType == provider.RelativeMetricType {
-				ch <- prometheus.MustNewConstMetric(
-					relativeDesc,
-					prometheus.GaugeValue,
-					data.Value,
-					data.Location,
-					e.node,
-					data.Provider,
-					e.region,
-					data.Units,
-				)
-			}
-		}
-	} else {
-		result, err := e.client.GetCarbonIntensity(ctx, e.location)
+	for _, data := range result {
+		desc, err := getMetricDesc(data)
```

Comment:
Looping through the results prevents the panic. 

I also took the chance to clean up the code a bit.